### PR TITLE
Feat: Cache npm dependencies #697

### LINKED ISSUE #697: Feature Request: Cache `npm` dependencies
**Is your feature request related to a problem? Please describe.**
We can cache the `node_modules` folder in workflows with artifacts to ensure faster workflow runs. 

**Describe the solution you'd like**
[Refer to this guide](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows)

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Harsh1s**
**What kind of change does this PR introduce?**

- [x] Features

**Issue Number:**

Fixes #697 

**Did you add tests for your changes?**

No

**Snapshots/Videos:**
N/A

**If relevant, did you update the documentation?**

No

**Summary**
Added caching for the `node_modules` folder in workflows with artifacts to ensure faster workflow runs.

Changes include:

1. Added the Cache node modules step before the existing Install dependencies step in Push-Workflow job in push-schema-doc.yml file.
2.  Added the Cache node modules step before the existing Install dependencies step in test-deploy job in documentation.yml file.
3. Modified the Install dependencies step to check for the cache hit.

**Does this PR introduce a breaking change?**

No

**Other information**

N/A

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-docs/blob/develop/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -20,7 +20,22 @@ jobs:
...
+          path: |
+            ~/.npm
+            node_modules
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
       - name: Install dependencies
+        if: ${{ steps.cache-npm.outputs.cache-hit != 'true' }}
```
Comment (EshaanAgg) [inline .github/workflows/documentation.yml]: Remove this conditional check.

File: .github/workflows/push-schema-doc.yml [hunk 2]
```diff
@@ -29,7 +29,23 @@ jobs:
...
+            ~/.npm
+            node_modules
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
       - name: Install dependencies
+        if: ${{ steps.cache-npm.outputs.cache-hit != 'true' }}
```
Comment (EshaanAgg) [inline .github/workflows/push-schema-doc.yml]: Remove this.

**Review by EshaanAgg — CHANGES_REQUESTED** [review]

File: .github/workflows/documentation.yml (hunk 1 above)
Comment (Harsh1s) [reply .github/workflows/documentation.yml]: Done, Thanks!

File: .github/workflows/push-schema-doc.yml (hunk 2 above)
Comment (Harsh1s) [reply .github/workflows/push-schema-doc.yml]: Done, appreciate the help!

**Comment by EshaanAgg** [comment]
@Harsh1s, it turns out this repository already uses `yarn cache` to cache the dependencies. Would have to close this PR as it would lead to redundancy. 
Sorry for the inconvenience.

**Pull request closed without merging**